backend/src/tables/tables.service.ts

```diff
@@ -172,13 +207,20 @@ export class TablesService {
 
         // Create a new table record if it doesn't exist and push the createTable promise into the promises array
         if (!existingTable) {
+          // Check to see if the LLM generated metadata for this table
+          const metadataForTable = _.find(
+            metadata,
+            (m) => m.name === tableName,
+          );
+
```

Comment:
Not urgent, but it may be more optimal to have metadata be an object where the keys are the table names, instead of an array. That way we can directly index into the relevant metadata instead of having to do a find each time. Would need to update the LLM response I think to conform with this change